lecture_5/homework/app/models/warrior.rb

```diff
@@ -14,4 +14,28 @@ class Warrior < ApplicationRecord
 
   scope :alive, -> { where('death_date IS NULL') }
   scope :dead, -> { where('death_date IS NOT NULL') }
-end
+
+  after_create :update_siege
+  before_update :update_prev_building
+  after_update  :update_siege_after
+  after_destroy :update_siege
+
+  def update_siege
+    Reports::SiegeReport.call(building: building) if building
+  end
+
+
+  def update_siege_after
+    if prev_building_id
+      Reports::SiegeReport.call(building: Building.find(prev_building_id))
+      Reports::SiegeReport.call(building: Building.find(building_id))
+      update_column(:prev_building_id, nil)
```

Comment:
Z tego co pamiętam update_column omija walidację więc nie jest to najlepszy sposób na edycje kolumny :D Najprosciej byłoby dodać relację typu belongs_to z opcją `optional: true` :) Optional pozwoli na ustawianie nila na tej kolumnie, nie trzeba nic zmieniać w bazie, bo bazy domyślnie pozwalają na nile, a jedynie rails sam waliduje to że nie mogą być nilem :)